x-pack/test/functional/services/ml/stack_management_jobs.ts

```diff
@@ -194,5 +198,99 @@ export function MachineLearningStackManagementJobsProvider(
+      });
+    },
+
+    async selectFileToImport(path: string, expectError: boolean = false) {
+      log.debug(`Importing file '${path}' ...`);
+      await PageObjects.common.setFileInputPath(path);
+
+      if (expectError) {
+        await testSubjects.existOrFail('~mlJobMgmtImportJobsFileReadErrorCallout');
+      } else {
+        await testSubjects.missingOrFail('~mlJobMgmtImportJobsFileReadErrorCallout');
+        await testSubjects.existOrFail('mlJobMgmtImportJobsFileRead');
+      }
+    },
+
+    async assertJobIdsExist(expectedJobIds: string[]) {
+      const inputs = await testSubjects.findAll('mlJobMgmtImportJobIdInput');
+      // const inputs = await subj.findAllByTagName('input');
+      const actualJobIds = await Promise.all(inputs.map((i) => i.getAttribute('value')));
+
+      expect(actualJobIds.sort()).to.eql(
+        expectedJobIds.sort(),
+        `Expected job ids to be '${JSON.stringify(expectedJobIds)}' (got '${JSON.stringify(
+          actualJobIds
+        )}')`
+      );
+    },
+
+    async assertCorrectTitle(jobCount: number, jobType: JobType) {
+      const dataTestSubj =
+        jobType === 'anomaly-detector'
+          ? 'mlJobMgmtImportJobsADTitle'
+          : 'mlJobMgmtImportJobsDFATitle';
+      const subj = await testSubjects.find(dataTestSubj);
+      const title = (await subj.parseDomContent()).html();
+
+      const jobTypeString =
+        jobType === 'anomaly-detector' ? 'anomaly detection' : 'data frame analytics';
+
+      const results = title.match(
+        /(\d) (anomaly detection|data frame analytics) job[s]? read from file$/
+      );
+      expect(results).to.not.eql(null, `Expected regex results to not be null`);
+      const foundCount = results![1];
+      const foundJobTypeString = results![2];
+      expect(foundCount).to.eql(
+        jobCount,
+        `Expected job count to be '${jobCount}' (got '${foundCount}')`
+      );
+      expect(foundJobTypeString).to.eql(
+        jobTypeString,
+        `Expected job count to be '${jobTypeString}' (got '${foundJobTypeString}')`
+      );
+    },
+
+    async assertJobIdsSkipped(expectedJobIds: string[]) {
+      const subj = await testSubjects.find('mlJobMgmtImportJobsCannotBeImportedCallout');
+      const skippedJobTitles = await subj.findAllByTagName('h5');
+      const actualJobIds = (
+        await Promise.all(skippedJobTitles.map((i) => i.parseDomContent()))
+      ).map((t) => t.html());
+
+      expect(actualJobIds.sort()).to.eql(
+        expectedJobIds.sort(),
+        `Expected job ids to be '${JSON.stringify(expectedJobIds)}' (got '${JSON.stringify(
+          actualJobIds
+        )}')`
+      );
+    },
+
+    async importJobs() {
+      await retry.tryForTime(5000, async () => {
+        await testSubjects.click('mlJobMgmtImportImportButton', 1000);
+      });
+      await retry.tryForTime(40000, async () => {
+        await testSubjects.missingOrFail('mlJobMgmtImportJobsFlyout');
+      });
```

Comment:
Is there a reason why you wrapped these in separate `retry` blocks?
The `click` with provided timeout already has a retry built in, so probably doesn't need another retry wrapper. And the `missingOrFail` can also take a parameter to specify the built in retry timeout, e.g.:
```typescript
await testSubjects.missingOrFail('mlJobMgmtImportJobsFlyout', { timeout: 40 * 1000 });
```
There is a case where we could need a wrapping retry around both calls: if we want to repeat the click in case the flyout doesn't go away, but not sure if we need that here.

Reply:
Updated in https://github.com/elastic/kibana/pull/110578/commits/9ac5c092dcc9b04de418a18ce43f0edfa93f032c